scripts/bin/opx-config-fanout

```diff
@@ -17,173 +17,223 @@ import cps
-                                               _g_if_name: if_name})
-        l.append(('rpc', cps_obj.get()))
-    tr = cps_utils.CPSTransaction(l)
-    tr.commit()
-
-    obj = cps_object.CPSObject(module='base-if-phy/breakout',
-                               data={
-                               'base-if-phy/breakout/input/front-panel-port':
-                                    fp_obj.get_attr_data(
-                                        'base-if-phy/front-panel-port/front-panel-port'),
-                               'base-if-phy/breakout/input/breakout-mode': _mode,
-                               'base-if-phy/breakout/input/port-speed': _speed
-                               })
-    tr = cps_utils.CPSTransaction([('rpc', obj.get())])
-
-    if tr.commit() == False:
-        print "Failed..."
-
-        # revert ports delete
+    for fp in fp_list:
+        fp_obj = nas_fp_cache.get(int(fp))
         l = []
-        # add ports to a transaction
+        npu = fp_obj.get_attr_data('npu-id')
+        phy_port_list = nas_if.physical_ports_for_front_panel_port(fp_obj)
         for phy_port in phy_port_list:
-            ifobj = nas_if.make_interface_from_phy_port(phy_port, current_mode, current_speed)
-            print "Creating interface " + ifobj.get_attr_data(_g_if_name)
-            l.append(('rpc', ifobj.get()))
-
-        # do the actual port create
+            if_port = if_cache.get_by_port(
+                    npu,
+                    phy_port.get_attr_data('port-id'))
+            if if_port is None:
+                continue
+            if_name = if_port.get_attr_data(_g_if_name)
+            print "Deleting.. %s" % if_name
+            cps_obj = cps_object.CPSObject(module = 'dell-base-if-cmn/set-interface',
+                                           data = {'dell-base-if-cmn/set-interface/input/operation': 2,
+                                                   _g_if_name: if_name})
+            l.append(('rpc', cps_obj.get()))
+    
         tr = cps_utils.CPSTransaction(l)
+        tr.commit()
+        print "tr.commit() status"
+        print "done deleting"
+        obj = cps_object.CPSObject(module='base-if-phy/breakout',
+                                   data={
+                                   'base-if-phy/breakout/input/front-panel-port':
+                                                fp_obj.get_attr_data(
+                                                        'base-if-phy/front-panel-port/front-panel-port'),
+                                   'base-if-phy/breakout/input/breakout-mode': _mode,
+                                   'base-if-phy/breakout/input/port-speed': _speed
+                                   })
+        tr = cps_utils.CPSTransaction([('rpc', obj.get())])
 
-        # commit
         if tr.commit() == False:
-            print "Failed to create interfaces"
+            print "Failed........."
```

Comment:
1. Be more descriptive about what the failure means here.
2. Be consistent about the number of dots you're using. One should be appropriate here.

Reply:
this print statement was in the original code